.travis.yml

```diff
@@ -1,81 +1,115 @@
+sudo: required
+dist: xenial
+
 language: go
 go:
   - "1.10.x"
   - "1.x"
 
-os: linux
-sudo: required
-dist: xenial
-
-services:
-  - docker
-
-addons:
-  artifacts: true
+cache:
+  directories:
+    - $TRAVIS_HOME/go
+    - $HOME/.gradle
 
 notifications:
   slack:
     secure: ZzIEqFE4XRdE9U2p3aeE32DMtoC8RgjoEavhEQ1oLrWFgUpLktqmp9UVY/U+W6iElilLpDbFpry51+Sv9MWpxJMxr+Q/JJuq/3Bj5KjF/wEtil7qvBYhQ1sM/qUQFG6wRkrMNjZGMiaTmnkWF0rZB8lf7+nbnGFaPW3AVVbD+8gVDWTHI4Hcvvgs0UbrJzoPfpvH0dprOchswc1BBKTgo5c44rvS2fquEMVcqMMiNJ5JQqphuRWLTfzLgOzImSf0/xJJyVp/YTkSnVSg8BcWmDCJ4iB9fJkVyZM9WxcgY/J4T5VzFxfMah9zv2j8UTfzHSMeCJDRL647hdnkmr/Qum/LN91Ey2DJw5KUH743CsAbyGhQML6wZ3NCeEP06hnMDphalU5+BYhtAPyc5CB84g6eLIUQ2EqptuPZpjFQohFnapCTnfB5XKTcW+PjxJsoJzk8x+85Xid+H1nnNxeyf10tLv6Pwy4ZGmEEbsa4SYWXibpIEu3fPJXEdtrht0vM40pDLeUYL6Axmh7hNjmDQOXJG41saF+Rk4AArRhKhMQTmlYCc0e1H2/hIDXUMPbqjHeCpEkaA5W8BFBKynhlJa0JX+rtHDFaK82Di8rXT0NO2ACyG8ZQqk87qePyBYPyfR8hRwhrkmQHlYYOZzV6LBz+ynJuWl9ktcC2irJlHZs=
 
-## Golang Test & CLI build
+
+before_install:
+  # Get Golang in Android environment
+  - if [ -z "$TRAVIS_GO_VERSION" ]; then eval "$(curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | GIMME_GO_VERSION=stable bash)"; fi
+  # Use cached directory as GOPATH
+  - export GOPATH=$TRAVIS_HOME/go
+  - export PATH=$GOPATH/bin:$PATH
+  - mkdir -p $GOPATH/src/github.com/sergeyfrolov
+  - mv $TRAVIS_BUILD_DIR $GOPATH/src/github.com/sergeyfrolov/gotapdance
```

Comment:
Because we want to cache the GOPATH. Caching restoration takes place before any of the stages, so it's best we use a GOPATH that's not the default directory Travis checkouts our code into.